File: SoloLearn Codes/Credit Card Validator/credit_card_validator.py
digits = [int(x) for x in input()[::-1]]
for i in range(1, len(digits), 2):
    digits[i] *= 2
    if digits[i] > 9:
        digits[i] -= 9

print("valid" if sum(digits) % 10 == 0 and len(digits) == 16 else "not valid")

File: SoloLearn Codes/Credit Card Validator/CreditCardValidator.cs
using System;
using System.Linq;

namespace SoloLearn
{
    class Program
    {
        static void Main(string[] args)
        {
            string input = Console.ReadLine();
            Console.WriteLine(input.Reverse()
                .Select((c, i) => i % 2 == 1 ? (int) char.GetNumericValue(c) * 2 : (int) char.GetNumericValue(c))
                .Select(x => x > 9 ? x - 9 : x)
                .Sum() % 10 == 0 && input.Length == 16 ? "valid" : "not valid");
        }
    }
}